pkg/artifactory/resource_artifactory_federated_repository_test.go

```diff
@@ -0,0 +1,125 @@
+package artifactory
+
+import (
+	"fmt"
+	"log"
+	"math/rand"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+func skipFederatedRepo() (bool, error) {
+	artifactory2Url := os.Getenv("ARTIFACTORY_URL_2")
+	testFederatedRepo := os.Getenv("ARTIFACTORY_TEST_FEDERATED_REPO")
+
+	if testFederatedRepo == "true" && len(artifactory2Url) > 0 {
+		log.Println("Both env var `ARTIFACTORY_TEST_FEDERATED_REPO` and `ARTIFACTORY_URL_2` are set. Executing test.")
+		return false, nil
+	}
+
+	log.Println("Either env var `ARTIFACTORY_TEST_FEDERATED_REPO` or `ARTIFACTORY_URL_2` is not set. Skipping test.")
+	return true, nil
+}
+
+func TestAccFederatedRepoWithMembers(t *testing.T) {
+	name := fmt.Sprintf("terraform-federated-generic-%d-full", rand.Int())
+	resourceType := "artifactory_federated_generic_repository"
+	resourceName := fmt.Sprintf("%s.%s", resourceType, name)
+	artifactoryUrl := os.Getenv("ARTIFACTORY_URL")
+	artifactory2Url := os.Getenv("ARTIFACTORY_URL_2")
+	federatedMemberUrl := fmt.Sprintf("%s/artifactory/%s", artifactoryUrl, name)
+	federatedMember2Url := fmt.Sprintf("%s/artifactory/%s", artifactory2Url, name)
+
+	const federatedRepositoryConfigFull = `
```

Comment:
I think this would be better using a template - The reason I started using templates is because it was hard to tell which param should be which